packages/flutter/lib/src/material/app_bar.dart

```diff
@@ -760,12 +760,18 @@ class AppBar extends StatefulWidget implements PreferredSizeWidget {
 }
 
 class _AppBarState extends State<AppBar> {
+  final Map<RawGestureDetector?, double> _notificationListenerStates = <RawGestureDetector?, double>{};
   ScrollNotificationObserverState? _scrollNotificationObserver;
   bool _scrolledUnder = false;
 
   @override
   void didChangeDependencies() {
     super.didChangeDependencies();
+    if (Scaffold.maybeOf(context)?.isDrawerOpen ?? false) {
+      _scrollNotificationObserver?.removeListener(_handleScrollNotification);
+      _scrollNotificationObserver = null;
+      return;
+    }
     _scrollNotificationObserver?.removeListener(_handleScrollNotification);
     _scrollNotificationObserver = ScrollNotificationObserver.maybeOf(context);
     _scrollNotificationObserver?.addListener(_handleScrollNotification);
```

Comment:
This could just be: 
```suggestion
    _scrollNotificationObserver?.removeListener(_handleScrollNotification);
    if (!Scaffold.maybeOf(context)?.isDrawerOpen ?? true) {
      _scrollNotificationObserver = ScrollNotificationObserver.maybeOf(context);
      _scrollNotificationObserver?.addListener(_handleScrollNotification);
    }
```